include/EASTL/intrusive_list.h

```diff
@@ -791,8 +791,8 @@ namespace eastl
 				EASTL_FAIL_MSG("intrusive_list::insert(): element already on a list.");
 		#endif
 
-		node_type& next = *const_cast<node_type*>(pos.mpNode);
-		node_type& prev = *static_cast<node_type*>(next.mpPrev);
+		intrusive_list_node& next = *const_cast<node_type*>(pos.mpNode);
```

Comment:
I'm not sure what you mean, do you mean line 795? I only changed from node_type& to intrusive_list_node&, I left the rest of the original code in tact.